components/scream/src/dynamics/homme/atmosphere_dynamics_fv_phys.cpp

```diff
@@ -0,0 +1,320 @@
+  if (not fv_phys_active()) return;
+  const auto& c = Homme::Context::singleton();
+  auto& gfr = c.get<Homme::GllFvRemap>();
+  const auto time_idx = c.get<Homme::TimeLevel>().n0;
+  constexpr int NGP = HOMMEXX_NP;
+  const int nelem = m_dyn_grid->get_num_local_dofs()/(NGP*NGP);
+  const auto npg = m_phys_grid_pgN*m_phys_grid_pgN;
+  const auto& gn = m_phys_grid->name();
+  const auto nlev = m_helper_fields.at("FT_phys").get_view<const Real**>().extent_int(1);
+  const auto nq = get_group_in("tracers", gn).m_bundle->get_view<const Real***>().extent_int(1);
+  assert(m_helper_fields.at("FT_phys").get_view<const Real**>().extent_int(0) == nelem*npg);
+
+  const auto uv_ndim = m_helper_fields.at("FM_phys").get_view<const Real***>().extent_int(1);
+  assert(uv_ndim == 2 or uv_ndim == 3);
+
+  const auto T = Homme::GllFvRemap::CPhys2T(
+    m_helper_fields.at("FT_phys").get_view<const Real**>().data(),
+    nelem, npg, nlev);
+  const auto uv = Homme::GllFvRemap::CPhys3T(
+    m_helper_fields.at("FM_phys").get_view<const Real***>().data(),
+    nelem, npg, uv_ndim, nlev);
+  const auto q = Homme::GllFvRemap::CPhys3T(
+    get_group_in("tracers", gn).m_bundle->get_view<const Real***>().data(),
+    nelem, npg, nq, nlev);
+  
+  gfr.run_fv_phys_to_dyn(time_idx, T, uv, q);
+  Kokkos::fence();
+  gfr.run_fv_phys_to_dyn_dss();
+  Kokkos::fence();
+}
+
+// [rrtmgp active gases] This is to address issue #1782. It supports option 1 in
+// that issue. These fv_phys_rrtmgp_active_gases_* routines can be removed once
+// rrtmgp active_gases initialization is treated properly.
+
+struct TraceGasesWorkaround {
+  bool restart{false};
+  std::shared_ptr<AbstractRemapper> remapper;
+  std::vector<std::string> active_gases; // other than h2o
+};
+
+static TraceGasesWorkaround s_tgw;
+
+void fv_phys_rrtmgp_active_gases_init (const ekat::ParameterList& p) {
+  const auto& v = p.sublist("atmosphere_processes").sublist("physics")
+    .sublist("rrtmgp").get<std::vector<std::string>>("active_gases");
+  for (const auto& e : v)
+    if (e != "h2o")
+      s_tgw.active_gases.push_back(e);
+}
+
+void fv_phys_rrtmgp_active_gases_set_restart (const bool restart) {
+  s_tgw.restart = restart;
+}
+
+void HommeDynamics
+::fv_phys_rrtmgp_active_gases_init (const std::shared_ptr<const GridsManager>& gm) {
+  if (s_tgw.restart) return; // always false b/c it hasn't been set yet
+  using namespace ekat::units;
+  using namespace ShortFieldTagsNames;
+  auto kgkg = kg/kg;
+  kgkg.set_string("kg/kg");
+  const auto& rgn = m_cgll_grid->name();
+  const auto& pgn = m_phys_grid->name();
+  const auto rnc = m_cgll_grid->get_num_local_dofs();
+  const auto pnc = m_phys_grid->get_num_local_dofs();
+  const auto nlev = m_cgll_grid->get_num_vertical_levels();
+  constexpr int ps = SCREAM_SMALL_PACK_SIZE;
+  for (const auto& e : s_tgw.active_gases) {
+    add_field<Required>(e, FieldLayout({COL,LEV},{rnc,nlev}), kgkg, rgn, ps);
+    // 'Updated' so that it gets written to the restart file.
```

Comment:
It is strange that we need "Updated" to ensure a write to the restart file.  My understanding is that "Required" should be enough, because that essentially says I need this field to be set before I run.  Does FV actually change the active gases so that it is indeed an "updated" variable?  If not and this is a hack to get it written in the restart file then there might be something we need to fix.  @bartgol any thoughts?

Reply:
I suspect it's b/c `Required` fields are read-only, while here we need to read the gases from a GLL grid and remap them to pg2. So on top of being Required (for restart reasons), the pg2 version needs to also be Computed (to be writable). Hence, Updated.

Reply:
Right. Required for restart + Computed b/c I'm mapping from CGLL IC input to pg2 for simulation. I can add to the comment to clarify this.